misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/OptionalAssertionsSpec.kt

```diff
@@ -1,45 +1,85 @@
 package ch.tutteli.atrium.specs.integration
 
 import ch.tutteli.atrium.api.fluent.en_GB.messageContains
+import ch.tutteli.atrium.api.fluent.en_GB.toBe
 import ch.tutteli.atrium.api.fluent.en_GB.toThrow
 import ch.tutteli.atrium.api.verbs.internal.expect
+import ch.tutteli.atrium.creating.Expect
 import ch.tutteli.atrium.specs.*
-import ch.tutteli.atrium.translations.DescriptionOptionalAssertion.EMPTY
+import ch.tutteli.atrium.translations.DescriptionOptionalAssertion
 import org.spekframework.spek2.Spek
 import org.spekframework.spek2.style.specification.Suite
 import java.lang.AssertionError
 import java.util.Optional
 
-abstract class OptionalAssertionsSpec (
-    isEmpty: Fun0<Optional<Any>>,
+abstract class OptionalAssertionsSpec(
+    isEmpty: Fun0<Optional<Int>>,
+    isPresentFeature: Feature0<Optional<Int>, Int>,
+    isPresent: Fun1<Optional<Int>, Expect<Int>.() -> Unit>,
+    isPresentNullableFeature: Feature0<Optional<Int?>, Int?>,
+    isPresentNullable: Fun1<Optional<Int?>, Expect<Int?>.() -> Unit>,
     describePrefix: String = "[Atrium] "
 ) : Spek({
 
-    include(object : SubjectLessSpec<Optional<Any>>(
+    include(object : SubjectLessSpec<Optional<Int>>(
         "$describePrefix[Path] ",
-        isEmpty.forSubjectLess()
+        isEmpty.forSubjectLess(),
+        isPresentFeature.forSubjectLess().adjustName { "$it feature" },
+        isPresent.forSubjectLess { toBe(1) }
+    ) {})
+    include(object : SubjectLessSpec<Optional<Int?>>(
+        describePrefix,
+        isPresentNullableFeature.forSubjectLess().adjustName { "$it feature" },
+        isPresentNullable.forSubjectLess { toBe(1) }
+    ) {})
+    include(object : AssertionCreatorSpec<Optional<Int>>(
+        describePrefix, Optional.of(2),
+        isPresent.forAssertionCreatorSpec("$toBeDescr: 2") { toBe(2) }
+    ) {})
+    include(object : AssertionCreatorSpec<Optional<Int?>>(
+        describePrefix, Optional.of(2) as Optional<Int?>,
```

Comment:
I am getting a warning here. Is it the right way to have an Optional of a nullable Int?

Reply:
Usually `Optional.of(2)` or `Optional.of<Int?>(2)` should be good enough. But I see my mistake here. I was in the wrong language, the value inside of Optional can never be null as Java has a requireNonNull check and Kotlin know it and thus turns even `Optional.of<Int?>(2)` into an `Optional<Int>`. 
Sorry for that but you can remove isPresentNullableFeature / isPresentNullable entirely